packages/did/src/Did.rpc.ts

```diff
@@ -5,104 +5,41 @@
-import type { DidDocument, KiltAddress, Service } from '@kiltprotocol/types'
-
+import type {
+  Did,
+  DidDocument,
+  DidUrl,
+  KiltAddress,
+  Service,
+  VerificationMethod,
+} from '@kiltprotocol/types'
 import { ss58Format } from '@kiltprotocol/utils'
-import { encodeAddress } from '@polkadot/keyring'
-import { ethereumEncode } from '@polkadot/util-crypto'
 
+import { fromChain, publicKeyFromChain, serviceFromChain } from './Did.chain.js'
+import { didKeyToVerificationMethod } from './Did.utils.js'
 import type {
   Address,
   SubstrateAddress,
 } from './DidLinks/AccountLinks.chain.js'
-import type {
-  ChainDidDetails,
-  ChainDidEncryptionKey,
-  ChainDidKey,
-  ChainDidVerificationKey,
-} from './Did.chain.js'
-
-import {
-  depositFromChain,
-  fragmentIdToChain,
-  fromChain,
-  publicKeyFromChain,
-} from './Did.chain.js'
-
-import { didKeyToVerificationMethod } from './Did.utils.js'
-import { addKeypairAsVerificationMethod } from './DidDetails/DidDetails.js'
-
-function documentFromChain(
-  encoded: DidDidDetails
-): Omit<ChainDidDetails, 'service'> {
-  const {
-    publicKeys,
-    authenticationKey,
-    attestationKey,
-    delegationKey,
-    keyAgreementKeys,
-    lastTxCounter,
-    deposit,
-  } = encoded
-
-  const keys: Record<string, ChainDidKey> = [...publicKeys.entries()]
-    .map(([keyId, keyDetails]) => publicKeyFromChain(keyId, keyDetails))
-    .reduce((res, key) => {
-      res[fragmentIdToChain(key.id)] = key
-      return res
-    }, {})
-
-  const authentication = keys[
-    authenticationKey.toHex()
-  ] as ChainDidVerificationKey
-
-  const didRecord: ChainDidDetails = {
-    authentication: [authentication],
-    lastTxCounter: lastTxCounter.toBn(),
-    deposit: depositFromChain(deposit),
-  }
-  if (attestationKey.isSome) {
-    const key = keys[attestationKey.unwrap().toHex()] as ChainDidVerificationKey
-    didRecord.assertionMethod = [key]
-  }
-  if (delegationKey.isSome) {
-    const key = keys[delegationKey.unwrap().toHex()] as ChainDidVerificationKey
-    didRecord.capabilityDelegation = [key]
-  }
-
-  const keyAgreementKeyIds = [...keyAgreementKeys.values()].map((keyId) =>
-    keyId.toHex()
-  )
-  if (keyAgreementKeyIds.length > 0) {
-    didRecord.keyAgreement = keyAgreementKeyIds.map(
-      (id) => keys[id] as ChainDidEncryptionKey
-    )
-  }
-
-  return didRecord
-}
-
-function serviceFromChain(encoded: DidServiceEndpointsDidEndpoint): Service {
-  const { id, serviceTypes, urls } = encoded
-  return {
-    id: `#${id.toUtf8()}`,
-    type: serviceTypes.map((type) => type.toUtf8()),
-    serviceEndpoint: urls.map((url) => url.toUtf8()),
-  }
-}
 
 function servicesFromChain(
-  encoded: DidServiceEndpointsDidEndpoint[]
+  encoded: DidServiceEndpointsDidEndpoint[],
+  did: Did
 ): Service[] {
-  return encoded.map((encodedValue) => serviceFromChain(encodedValue))
+  return encoded.map((encodedValue) => {
+    const service = serviceFromChain(encodedValue)
```

Comment:
if we don't want to change the implementation of serviceToChain, I'd change this to
```suggestion
    const service = serviceFromChain(new Option(encodedValue))
```

Reply:
turns out it's not that simple. We could do:
```suggestion
    const service = serviceFromChain({
      unwrap: () => encodedValue,
    })
```
But it would get a little hacky. Maybe we just stick to accepting both options and plain service records on the decoder.

Reply:
@ntn-x2 This is too hacky for my taste so I suggest sticking with accepting both `Option<DidServiceEndpointsDidEndpoint>` & `DidServiceEndpointsDidEndpoint` on serviceFromChain, which is easily implemented. Agreed?

Reply:
Yeah let's go with that then. Do you want to apply the same changes to other places that accept an `Option<T>` as well?

Reply:
I was thinking about it... it would be more consistent wouldn't it?

Reply:
done in d7bebbf66800cdd99f3660dfb312761545a4e7f6